docs/loading/Flink-connector-starrocks.md

```diff
@@ -1,173 +1,647 @@
+            .withProperty("database-name", "test")
+            .withProperty("table-name", "score_board")
+            .withProperty("username", "root")
+            .withProperty("password", "")
+            .build();
+
+    // connector will use a Java object array (Object[]) to represent a row of
+    // StarRocks table, and each element is the value for a column. Need to
+    // define the schema of the Object[] which matches that of StarRocks table
+    TableSchema schema = TableSchema.builder()
+            .field("id", DataTypes.INT().notNull())
+            .field("name", DataTypes.STRING())
+            .field("score", DataTypes.INT())
+            // Must specify the primary key for StarRocks primary key table,
+            // and DataTypes.INT().notNull() for `id` must specify notNull()
+            .primaryKey("id")
+            .build();
+    // Transforms the RowData to the Object[] according to the schema
+    RowDataTransformer transformer = new RowDataTransformer();
+    // Create the sink with schema, options, and transformer
+    SinkFunction<RowData> starRockSink = StarRocksSink.sink(schema, options, transformer);
+    source.addSink(starRockSink);
+    ```
+
+  - The `RowDataTransformer` in the main program is defined as
+
+    ```java
+    private static class RowDataTransformer implements StarRocksSinkRowBuilder<RowData> {
+    
+        /**
+         * Set each element of the object array according to the input RowData.
+         * The schema of the array matches that of StarRocks table.
+         */
+        @Override
+        public void accept(Object[] internalRow, RowData rowData) {
+            internalRow[0] = rowData.id;
+            internalRow[1] = rowData.name;
+            internalRow[2] = rowData.score;
+            // Only need for StarRocks primary key table. Set the last
+            // element to tell whether the record is a UPSERT or DELETE.
+            internalRow[internalRow.length - 1] = StarRocksSinkOP.UPSERT.ordinal();
+        }
+    }  
+    ```
+
+## Best Practices
+
+### Load data to primary key table
+
+This section will show how to load data to StarRocks primary key table to achieve partial update, and conditional update.
+You can see [Change data through loading](https://docs.starrocks.io/en-us/latest/loading/Load_to_Primary_Key_tables) for the introduction of those features.
+These examples use Flink SQL.
+
+#### Preparations
+
+Create a database `test` and create a Primary Key table `score_board` in StarRocks.
+
+```SQL
+CREATE DATABASE `test`;
+
+CREATE TABLE `test`.`score_board`
+(
+    `id` int(11) NOT NULL COMMENT "",
+    `name` varchar(65533) NULL DEFAULT "" COMMENT "",
+    `score` int(11) NOT NULL DEFAULT "0" COMMENT ""
+)
+ENGINE=OLAP
+PRIMARY KEY(`id`)
+COMMENT "OLAP"
+DISTRIBUTED BY HASH(`id`);
+```
+
+#### Partial update
+
+This example will show how to load data only to columns `id` and `name`.
+
+1. Insert initial data to StarRocks table in MySQL client
```

Comment:
![image](https://github.com/StarRocks/starrocks/assets/57167462/cb4e35f3-767b-4569-a037-13bc849336ce)
-> Insert to the StarRocks table `score_board` in your MySQL client.